app/controllers/reviews_controller.rb

```diff
@@ -33,23 +33,27 @@ def edit
   def create
     @review = Review.new(review_params)
 
-    respond_to do |format|
-      if @review.save
-        format.html do
-          redirect_to professor_path(@review.professor_id),
-                      notice: 'Review was successfully created.'
-        end
-        format.json { render :show, status: :created, location: @review }
-      else
-        format.html do
-          @courses = load_courses
-          @professors = load_professors
-          render :new
-        end
-        format.json do
-          render json: @review.errors, status: :unprocessable_entity
+    if (params[:course_id] && params[:course_id] != '') &&
+         (params[:professor_id] && params[:professor_id] != '')
+      @review.course_id = params[:course_id] if @review.course_id.nil?
+      @review.professor_id = params[:professor_id] if @review.professor_id.nil?
```

Comment:
```suggestion
      @review.course_id = params[:course_id]
      @review.professor_id = params[:professor_id]
```
They're guaranteed to not be `nil` or `''` at this point.

Reply:
I thought they would be guaranteed to not be "" or nil as well but when testing through capybara, if a professor or course isn't selected, then their respective id will be ' '. While the frontend prevents user from leaving those fields blank, if they send a post request using Postman or a similar app, without interacting with the frontend, the review object will be sent to the controller without the ids, thus breaking the code.